python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -0,0 +1,364 @@
+        xmin, xmax = bbox.getMinX(), bbox.getMaxX()
+        ymin, ymax = bbox.getMinY(), bbox.getMaxY()
+        bounds = galsim.bounds.BoundsI(xmin, xmax, ymin, ymax)
+
+        image = galsim.Image(exposure.image[bbox].array, bounds=bounds, dtype=float, copy=False)
+        psf = galsim.Image(psfImage.array, dtype=float, copy=False)
+        # FIXME: no bounds in the line above? apparently no bounds in cpp
+
+        # psfImage = psf  #galsim.Image(psf[bbox].array, bounds=bounds, copy=False)
+
+        # Get the `lsst.meas.base` mask for bad pixels.
+        subMask = exposure.mask[bbox]
+        # FIXME: PARENT? we want subMask(*afwMask, bbox, afw::image::PARENT);
+        # where afwMask=exposure.mask
+        badpix = subMask.array.copy()
+        bitValue = exposure.mask.getPlaneBitMask(self.config.badMaskPlanes)
+        badpix &= bitValue
+
+        # # Convert an afw mask into a GalSim image for use with HSM. HSM uses a
+        # # mask where 0 = bad, 1 = good.
+        # hsmMask = galsim.ImageI(subMask.array, badpix=badpix, copy=False)
+        # mask = hsmMask[bbox]
+
+        # Convert the mask of bad pixels to a format suitable for galsim.
+        # NOTE: galsim.Image will match whatever dtype the input array is
+        # (here int32).
+        badpix = galsim.Image(badpix, bounds=bounds, copy=False)
+        # mask = hsmMask #[bbox]
+
+        # FIXME: dummyMask not used later on?! (it was in cpp!)
+        # dummyMask = galsim.ImageI(bounds=bounds)
+        # dummyMask.image = 1  # or something to make all pixel values 1
+        # # psfMask = dummyMask
+
+        sctrl = afwMath.StatisticsControl()
+        sctrl.setAndMask(bitValue)
+
+        # NOTE: Origin defaults to PARENT in all cases accessible from Python.
+        variance = afwImage.Image(
+            exposure.variance[bbox],
+            dtype=psfImage.dtype,
+            deep=True,
+        )
+
+        stat = afwMath.makeStatistics(variance, subMask, afwMath.MEDIAN, sctrl)
+
+        skyvar = stat.getValue(afwMath.MEDIAN)
+
+        # shape = galsim.hsm.ShapeData()
+        # psfShape = galsim.hsm.ShapeData()
+
+        try:
+            shape = galsim.hsm.EstimateShear(
+                gal_image=image,
+                PSF_image=psf,
+                weight=None,
+                badpix=badpix,
+                sky_var=skyvar,
+                shear_est=self.config.shearType,
+                recompute_flux="FIT",
+                guess_sig_gal=2.5 * psfSigma,
+                guess_sig_PSF=psfSigma,
+                precision=1.0e-6,
+                guess_centroid=galsim.PositionD(center.getX(), center.getY()),
+                strict=True,
+                hsmparams=None,
+            )
+        except galsim.hsm.GalSimHSMError as error:
+            raise measBase.MeasurementError(str(error), self.GALSIM.number)
+
+        assert shape.meas_type == self.measTypeSymbolLookup[self.config.measType]
```

Comment:
On second thoughts, I don't think you should be checking this here at all. This does not depend on the source/object we are measuring and is wasteful to check this for each measurement. We should check that this is true in one of the unit tests and be done with it.

Reply:
The unit test could call `FindAdaptiveMom` to get a `ShapeData` object and check this.

Reply:
I believe you're referring to EstimateShear(). Initially, I tried to avoid it due to the numerous values required for the call. However, I eventually implemented a minimal version in the unit test, which suffices to determine the measurement type outputted by GalSim after the shear estimation process.